utils/v2tofhirr4/mapping_defs.bal

```diff
@@ -0,0 +1,131 @@
+public type Pv1ToPatient function (Pv1 pv1) returns r4:Patient;
+
+# Mapping function type for PV1 segment to Encounter FHIR resource.
+public type Pv1ToEncounter function (Pv1 pv1) returns r4:Encounter;
+
+# Mapping function type for NK1 segment to Patient FHIR resource.
+public type Nk1ToPatient function (Nk1 nk1) returns r4:Patient;
+
+# Mapping function type for PD1 segment to Patient FHIR resource.
+public type Pd1ToPatient function (Pd1 pd1) returns r4:Patient;
+
+# Mapping function type for PID segment to Patient FHIR resource.
+public type PidToPatient function (Pid pid) returns r4:Patient;
+
+# Mapping function type for DG1 segment to Condition FHIR resource.
+public type Dg1ToCondition function (Dg1 dg1) returns r4:Condition;
+
+# Mapping function type for OBX segment to Observation FHIR resource.
+public type ObxToObservation function (Obx obx) returns r4:Observation;
+
+# Mapping function type for OBR segment to DiagnosticReport FHIR resource.
+public type ObrToDiagnosticReport function (Obr obr) returns r4:DiagnosticReport;
+
+# Mapping function type for AL1 segment to AllergyIntolerance FHIR resource.
+public type Al1ToAllerygyIntolerance function (Al1 al1) returns r4:AllergyIntolerance;
+
+# Mapping function type for EVN segment to Provenance FHIR resource.
+public type EvnToProvenance function (Evn evn) returns r4:Provenance;
+
+# Mapping function type for MSH segment to MessageHeader FHIR resource.
+public type MshToMessageHeader function (Msh msh) returns r4:MessageHeader;
+
+# Mapping function type for PV2 segment to Encounter FHIR resource.
+public type Pv2ToEncounter function (Pv2 pv2) returns r4:Encounter;
+
+# Mapping function type for ORC segment to Immunization FHIR resource.
+public type OrcToImmunization function (Orc orc) returns r4:Immunization;
+
+# V2toFHIR Mapper function implementation holder record.
+#
+# + pv1ToPatient - PV1 segment to Patient FHIR resource mapping function  
+# + pv1ToEncounter - PV1 segment to Encounter FHIR resource mapping function  
+# + nk1ToPatient - NK1 segment to Patient FHIR resource mapping function 
+# + pd1ToPatient - PD1 segment to Patient FHIR resource mapping function  
+# + pidToPatient - PID segment to Patient FHIR resource mapping function  
+# + dg1ToCondition - DG1 segment to Condition FHIR resource mapping function  
+# + obxToObservation - OBX segment to Observation FHIR resource mapping function  
+# + obrToDiagnosticReport - OBR segment to DiagnosticReport FHIR resource mapping function  
+# + al1ToAllerygyIntolerance - AL1 segment to AllergyIntolerance FHIR resource mapping function  
+# + evnToProvenance - EVN segment to Provenance FHIR resource mapping function  
+# + mshToMessageHeader - MSH segment to MessageHeader FHIR resource mapping function  
+# + pv2ToEncounter - PV2 segment to Encounter FHIR resource mapping function  
+# + orcToImmunization - ORC segment to Immunization FHIR resource mapping function
+public type V2toFhirMapperImpl record {
```

Comment:
can we make this a closed record?

Reply:
The idea is to provide user to inject their own functions as well. Is there a way to handle that kind of scenario with closed record?

Reply:
ah ok got it..